pilot/pkg/model/push_context.go

```diff
@@ -1612,7 +1612,23 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) error {
 		return err
 	}
 
-	sortConfigByCreationTime(envoyFilterConfigs)
+	sort.SliceStable(envoyFilterConfigs, func(i, j int) bool {
+		ifilter := envoyFilterConfigs[i].Spec.(*networking.EnvoyFilter)
+		jfilter := envoyFilterConfigs[j].Spec.(*networking.EnvoyFilter)
+		// If prirority is same fallback to name and creation timestamp, else use prirority.
+		if ifilter.Priority == jfilter.Priority {
+			// If creation time is the same, then behavior is nondeterministic. In this case, we can
+			// pick an arbitrary but consistent ordering based on name and namespace, which is unique.
+			// CreationTimestamp is stored in seconds, so this is not uncommon.
+			if envoyFilterConfigs[i].CreationTimestamp == envoyFilterConfigs[j].CreationTimestamp {
+				in := envoyFilterConfigs[i].Name + "." + envoyFilterConfigs[i].Namespace
+				jn := envoyFilterConfigs[j].Name + "." + envoyFilterConfigs[j].Namespace
+				return in < jn
+			}
```

Comment:
Can you remove this, we donot sort any other resources by name/ns

Reply:
We do. look at https://github.com/istio/istio/blob/aee75f9a4badbde6822c20657c01764ff8c5ba08/pilot/pkg/model/config.go#L342

Reply:
fair enough

Reply:
I think instead of triple nested, better to early evaluate like the following 
if Ipriority != jpriority { return ipriorty < jpriory}